solana/system_program.py

```diff
@@ -304,14 +335,50 @@ def sys_program_id() -> PublicKey:
     return PublicKey("11111111111111111111111111111111")
 
 
-def create_account(param: CreateAccountParams) -> Transaction:
+def create_account(params: CreateAccountParams) -> Transaction:
     """Generate a Transaction that creates a new account."""
-    raise NotImplementedError("create_account not implemented")
+    layout = SYSTEM_INSTRUCTION_LAYOUTS[CREATE_IDX]
+    data = encode_data(layout, params.lamports, params.space, params.program_id.__bytes__())
+
+    txn = Transaction()
+    txn.add(
+        TransactionInstruction(
+            keys=[
+                AccountMeta(pubkey=params.from_pubkey, is_signer=True, is_writable=True),
+                AccountMeta(pubkey=params.new_account_pubkey, is_signer=False, is_writable=True),
+            ],
+            program_id=sys_program_id(),
+            data=data,
+        )
+    )
+    return txn
 
 
 def assign(params: Union[AssignParams, AssignWithSeedParams]) -> Transaction:
     """Generate a Transaction that assigns an account to a program."""
-    raise NotImplementedError("assign not implemented")
+    if hasattr(params, "base_pubkey"):
+        params = cast(AssignWithSeedParams, params)
+        data = encode_data(
+            SYSTEM_INSTRUCTION_LAYOUTS[ASSIGN_WITH_SEED_IDX],
```

Comment:
This isn't going to work here b/c I left the format for all `SEED` based instructions blank:

https://github.com/michaelhly/solana-py/blob/d636be38aba6d7543b494335a18e904d43024c5a/solana/system_program.py#L211

https://github.com/solana-labs/solana-web3.js/blob/093a786bc8c73be83d05b28408450be87232806b/src/layout.js#L22

Looks like the SEED can have a variable-length? We might just pack everything manually with a `bytearray` instead of using the `Struct` class.

Reply:
Yes, you're right. The parsing speed is a bit complicated since we need to first get the length of the string.

Reply:
Currently, my idea is to construct the format dynamically and solve this. Let me see if this will work.

Reply:
Maybe we can skip this and include it in a separate PR?

Reply:
sounds good.